department-of-veterans-affairs/va.gov-cms
# VACMS-10626: Drupalize new homepage components

## Description

Closes #10626 

New Block Types:
Benefit Promo
News Promo

New entityqueues:
Home page news spotlight
Home page hero

New Menus:
Other Search Tools
Popular on VA.gov

## QA steps

- [ ] Create benefit promo block: https://pr11016-b9g0h6rk9avw9mnrfirx6ypjsgssems7.ci.cms.va.gov/block/add/benefit_promo
- [ ] Create news promo block: https://pr11016-b9g0h6rk9avw9mnrfirx6ypjsgssems7.ci.cms.va.gov/block/add/news_promo
- [ ] Add new benefit promo block to hero entityqueue: https://pr11016-b9g0h6rk9avw9mnrfirx6ypjsgssems7.ci.cms.va.gov/admin/structure/entityqueue/home_page_hero/home_page_hero?destination=/admin/structure/entityqueue
- [ ] Add new news promo block to spotlight entityqueue: https://pr11016-b9g0h6rk9avw9mnrfirx6ypjsgssems7.ci.cms.va.gov/admin/structure/entityqueue/home_page_news_spotlight/home_page_news_spotlight?destination=/admin/structure/entityqueue
- [ ] Add one or more links to other-search-tools menu: https://pr11016-b9g0h6rk9avw9mnrfirx6ypjsgssems7.ci.cms.va.gov/admin/structure/menu/manage/other-search-tools
Add one or more links to the popular-on-va-gov menu: https://pr11016-b9g0h6rk9avw9mnrfirx6ypjsgssems7.ci.cms.va.gov/admin/structure/menu/manage/popular-on-va-gov

Run GraphQL queries:
Validate GraphQL queries provide expected output:

```graphql
query homePageHeroQuery {
  entitySubqueueById(id: "home_page_hero") {
    ... on EntitySubqueueHomePageHero {
      itemsOfEntitySubqueueHomePageHero {
        entity {
          entityId
          ... on BlockContentBenefitPromo {
            entityId
            entityLabel
            fieldPromoHeadline
            fieldPromoText
            fieldPromoCta {
              entity {
                ... on ParagraphButton {
                  fieldButtonLink {
                    url {
                      path
                    }
                  }
                  fieldButtonLabel
                }
              }
            }
          }
        }
      }
    }
  }
}


query homePageNewsSpotlightQuery {
  entitySubqueueById(id: "home_page_news_spotlight") {
    ... on EntitySubqueueHomePageNewsSpotlight {
      itemsOfEntitySubqueueHomePageNewsSpotlight {
        entity {
          entityId
          ... on BlockContentNewsPromo {
            entityId
            entityLabel
            fieldPromoHeadline
            fieldPromoText
            fieldPromoLink {
              entity {
                ... on ParagraphLinkTeaser {
                  fieldLink {
                    url {
                      path
                    }
                    title
                  }
                }
              }
            }
          }
        }
      }
    }
  }
}


query homePageOtherSearchToolsMenuQuery {
  menuByName(name: "other-search-tools") {
    name
    links {
      label
      url {
        path
      }
      links {
        label
        url {
          path
        }
      }
    }
  }
}

query homePagePopularOnVaGovMenuQuery {
  menuByName(name: "popular-on-va-gov") {
    name
    links {
      label
      url {
        path
      }
      links {
        label
        url {
          path
        }
      }
    }
  }
}
```


### Definition of Done




### Select Team for PR review



### Is this PR blocked by another PR?


### Does this PR need review from a Product Owner


### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
